### DISCUSSION TITLE
Cannot redefine property: $persist

### DISCUSSION THREAD

**Discussion by morawcik** [livewire/livewire#8837] [1. Help]
Hi,

A few of my users are getting this error.
I'm only using simple `@livewireScripts` without any additional alpine include/load etc etc

**Comment by jacobgraf** [comment]
Experiencing this as well on Livewire v4 with Laravel 12. The error fires on back/forward browser
  navigation due to DisableBackButtonCacheMiddleware.

  Setup:
  - Alpine imported from Livewire's bundled ESM (import { Livewire, Alpine } from
  '../../vendor/livewire/livewire/dist/livewire.esm')
  - No separate Alpine installation (only @alpinejs/intersect as an extra plugin)
  - Laravel Herd local dev environment

  Behavior:
  - Uncaught TypeError: Cannot redefine property: $persist fires in app.js when the browser restores a
  cached page
  - No functional impact on the app itself, but the console error triggers Laravel Herd's error overlay
  to pop up on every back-navigation, which is disruptive during development. Herd surfaces JS errors
  automatically, so this creates a constant interruption when navigating around the app.
  - Consistent and reproducible across Chrome and Safari

  This isn't a duplicate script loading issue since Alpine is only loaded once through Livewire's bundle.
   The root cause appears to be the back-button cache middleware causing the page to re-initialize
  scripts without a full reload, which tries to re-register Alpine plugins that are already defined.

  Would love to see this addressed, even if it's just a guard check (if
  (!Object.getOwnPropertyDescriptor(Alpine, '$persist'))) before registering the plugin. Given that Herd
  is the recommended Laravel dev environment, this is a pretty rough DX issue for anyone using both
  together as it causes the Herd console to pop open when it triggers.

**Reply by joshhanley** [reply]
@jacobgraf any chance you could provide us a small example repo that demonstrates the issue, with step by step instructions on how to replicate it so we can investigate? Also what is the Herd error overlay? I use Herd and have never seen that. Is that a setting you have enabled?
